frontend/src/v5/store/revisions/revisions.redux.ts

```diff
@@ -40,106 +41,68 @@ export const INITIAL_STATE: IRevisionsState = {
 	revisionsUploadStatus: {},
 };
 
-export const setVoidStatusSuccess = (state = INITIAL_STATE, {
+export const setVoidStatusSuccess = (state, {
 	containerId,
 	revisionId,
 	isVoid,
-}): IRevisionsState => ({
-	...state,
-	revisionsByContainer: {
-		...state.revisionsByContainer,
-		[containerId]: state.revisionsByContainer[containerId].map((revision) => ({
-			...revision,
-			void: [revision.tag, revision._id].includes(revisionId) ? isVoid : revision.void,
-		})),
-	},
-});
-
-export const fetchSuccess = (state = INITIAL_STATE, {
+}: SetRevisionVoidStatusSuccessAction) => {
+	const revisions = state.revisionsByContainer[containerId];
+	revisions.find(
+		(revision) => [revision.tag, revision._id].includes(revisionId),
+	).void = isVoid;
+};
+
+export const fetchSuccess = (state, {
 	containerId,
 	revisions,
-}): IRevisionsState => ({
-	...state,
-	revisionsByContainer: {
-		...state.revisionsByContainer,
-		[containerId]: revisions,
-	},
-});
-
-export const updateRevisionSuccess = (state = INITIAL_STATE, {
+}: FetchSuccessAction) => {
+	state.revisionsByContainer[containerId] = revisions;
+};
+
+export const updateRevisionSuccess = (state, {
 	containerId,
 	data,
-}): IRevisionsState => ({
-	...state,
-	revisionsByContainer: {
-		...state.revisionsByContainer,
-		[containerId]: [
-			...state.revisionsByContainer[containerId].map((revision) => {
-				if (revision._id === data._id) return { ...revision, ...data };
-				return revision;
-			}),
-		],
-	},
-});
-
-export const setIsPending = (state = INITIAL_STATE, { isPending, containerId }): IRevisionsState => ({
-	...state,
-	isPending: {
-		...state.isPending,
-		[containerId]: isPending,
-	},
-});
-
-export const setUploadComplete = (state = INITIAL_STATE, {
+}: UpdateRevisionSuccessAction) => {
+	const revisions = state.revisionsByContainer[containerId];
+	const index = revisions.findIndex(({ _id }) => _id === data._id);
+	revisions[index] = { ...revisions[index], ...data };
+};
+
+export const setIsPending = (state, { isPending, containerId }: SetIsPendingAction) => {
+	state.isPending[containerId] = isPending;
+};
+
+export const setUploadComplete = (state, {
 	uploadId,
 	isComplete,
 	errorMessage,
-}): IRevisionsState => ({
-	...state,
-	revisionsUploadStatus: {
-		...state.revisionsUploadStatus,
-		[uploadId]: {
-			...state.revisionsUploadStatus[uploadId],
-			isComplete,
-			errorMessage,
-		},
-	},
-});
-
-export const revisionProcessingSuccess = (state = INITIAL_STATE, {
+}: SetUploadCompleteAction) => {
+	const uploads = state.revisionsUploadStatus;
+	uploads[uploadId] = { ...(uploads[uploadId] || {}), isComplete, errorMessage };
```

Comment:
I tried doing this in the console
````
var uploads = {}
var pp = {...uploads['2323'], isComplete:true}
````
And it didnt cause an error... 🤷‍♂️


```suggestion
	uploads[uploadId] = { ...uploads[uploadId], isComplete, errorMessage };
```

Reply:
uploads may be `undefined` and I thought spreading it caused issues, but it doesn't, so I changed the code as you suggested